### PR TITLE
Ticket Form Selector

### PR DESCRIPTION
**Pull request by Fredx87**
## Description

This PR adds a ticket form selector developed in JavaScript using React and Garden components.

It also introduces a new folder `src/modules` which contains JS/JSX files that will be bundled using ES module format. It is possible to import these modules from the Curlybars template using a `<script type="module">` tag.
The bundles created with this setup are emitted as assets, and since they are renamed when deployed, an import map is needed to retrieve the correct file in the theme.

React and Garden components are imported from NPM, and the Rollup setup has been extended to support the compilation of the modules, condensing all the dependencies in a separate bundle, and customizing the generated code to make import maps work. This setup will be optimized in the future.

## Screenshots
Screenshot 2023-07-17 at 16 45 52

## Checklist

- [ ] :green_book: all commit messages follow the [conventional commits](https://conventionalcommits.org/) standard
- [ ] :arrow_left: changes are compatible with RTL direction
- [ ] :wheelchair: Changes to the UI are [tested for accessibility](./../README.md#accessibility-testing) and compliant with [WCAG 2.1](https://www.w3.org/TR/WCAG21/).
- [ ] :memo: changes are tested in Chrome, Firefox, Safari and Edge
- [ ] :iphone: changes are responsive and tested in mobile
- [ ] :+1: PR is approved by @zendesk/vikings

### CODE REVIEW

File: templates/new_request_page.hbs [hunk 1]
```diff
@@ -18,7 +18,16 @@
...
+  <div id="main-content" class="form" data-props='
+  {
+    "ticketForms": {
+     "options": [
+      {{ #each ticket_forms }}
+       { "id": {{id}}, "url": {{json_stringify url}}, "name": {{json_stringify display_name}}  } {{#unless @last}},{{/unless}}
+      {{ /each }}
+     ]
+    }
+  }
```
Comment (anpa) [inline templates/new_request_page.hbs:30]: Another way of exposing this data is:

```hbs
<script>
  var forms = []
  {{#each ticket_forms}}
    forms.push({
      id: {{id}},
      url: "{{url}}",
      displayName: "{{display_name}}"
    })
  {{/each}}
  
  console.log(forms);
</script>


```

**Review by sergioazevedo — APPROVED** [review]

File: rollup.config.mjs [hunk 2]
```diff
@@ -1,13 +1,84 @@
...
+    plugins: [zass()],
+    watch: {
+      clearScreen: false,
+    },
   },
-  plugins: [zass()],
-  watch: {
-    clearScreen: false,
+  {
+    input: ["src/modules/NewRequestForm.jsx", "src/modules/shared.jsx"],
```
Comment (luis-almeida) [inline rollup.config.mjs:21]: Would be nice to be consistent in terms of naming.
It doesn't seem like we're following the pattern of having the file name aligned with the default exported function name so I'm wondering if it makes sense to have some bundles in camel case and some other not.

**Review by luis-almeida — COMMENTED** [review]
Looks really good! Just left a couple of comments/questions.

There's things we can address in follow up PRs, e.g. polish import maps or compilation of `assets` after merging.

File: rollup.config.mjs (hunk 2 above)
Comment (Fredx87) [reply rollup.config.mjs:21]: It makes sense to have consistent file naming. I think initially I just exported the component so I called the file with the component name, but now we export a function, so probably it makes sense to change the file name as well

**Pull request merged**